### PR TITLE
handle panic while registering new relic

### PR DESCRIPTION
**Pull request by dineshgoyal092**

### CODE REVIEW

File: src/nr_init.rs [hunk 1]
```diff
@@ -12,7 +13,23 @@ lazy_static! {
...
+                Ok(_) => true,
+                Err(_) => {
+                    println!("error_in_new_relic_integration");
+                    false
+                }
+            };
+            flag
+        } else {
+            false
+        }
```
Comment (amitu) [inline src/nr_init.rs:33]: You don't need the else block, you are not assigning the result of `if` condition on anything.

File: src/nr_init.rs [hunk 2]
```diff
@@ -12,7 +13,23 @@ lazy_static! {
...
+            let result = panic::catch_unwind(|| {
+                NR_APP.status()
+            });
+            let flag = match result {
+                Ok(_) => true,
+                Err(_) => {
+                    println!("error_in_new_relic_integration");
+                    false
+                }
+            };
```
Comment (amitu) [inline src/nr_init.rs:29]:
```suggestion
            if panic::catch_unwind(|| {
                NR_APP.status()
            }).is_err() {
                    println!("error_in_new_relic_integration");
            };
```

File: src/nr_init.rs [hunk 3]
```diff
@@ -12,7 +13,23 @@ lazy_static! {
     };
     pub static ref ENABLE_NEW_RELIC: bool = {
         let enable_nr = env::var("ENABLE_NEW_RELIC").unwrap_or_else(|_| "false".to_string());
+        //        println!("enable_nr_flag_value {:?}", &enable_nr);
         let x: bool = FromStr::from_str(&enable_nr).unwrap();
-        x
+        if x
```
Comment (amitu) [inline src/nr_init.rs:18]:
```suggestion
        if env::var("ENABLE_NEW_RELIC") == Some("true") {
```

Doesn't work?

**Review by amitu — APPROVED** [review]
Looks good to me.

**Comment by dineshgoyal092** [comment]
thanks for reviewing @amitu

File: src/nr_init.rs [hunk 4]
```diff
@@ -11,8 +12,17 @@ lazy_static! {
         app
     };
     pub static ref ENABLE_NEW_RELIC: bool = {
-        let enable_nr = env::var("ENABLE_NEW_RELIC").unwrap_or_else(|_| "false".to_string());
-        let x: bool = FromStr::from_str(&enable_nr).unwrap();
-        x
+        if env::var("ENABLE_NEW_RELIC").unwrap_or_else(|_| "false".to_string())
+            == "true".to_string()
```
Comment (amitu) [inline src/nr_init.rs:16]:
```suggestion
        if env::var("ENABLE_NEW_RELIC") == Some("true".to_string())
```

**Pull request merged**